cmd/nerdctl/network_rm.go

```diff
@@ -91,16 +92,32 @@ func networkRmAction(cmd *cobra.Command, args []string) error {
 		},
 	}
 
+	status := 0
 	for _, name := range args {
 		if name == "host" || name == "none" {
-			return fmt.Errorf("pseudo network %q cannot be removed", name)
+			status = 1
+			logrus.Errorf("pseudo network %q cannot be removed", name)
+			continue
 		}
 
 		n, err := walker.Walk(cmd.Context(), name)
 		if err != nil {
-			return err
+			status = 1
+			logrus.Error(err)
+			continue
+
 		} else if n == 0 {
-			return fmt.Errorf("no such network %s", name)
+			status = 1
+			logrus.Errorf("No such network: %s", name)
+			continue
+		}
+	}
+
+	// compatible with docker
+	if status != 0 {
+		return ExitCodeError{
```

Comment:
return `errors.New(..)` is enough

Reply:
`ExitCodeError` is to allow the program to exit with status code 1 without outputting an error message.

if return `errors.New(..)`, will output redundant error lines, like 
```bash
ERRO[0000] network "deleteme" is in use by container ["182ea93ddc15b187419a05cb10a4b8d3cdace1fc2b8592a10b345396ce91edc7"] 
FATA[0000] 
```

Reply:
There is no error message needs to be output here(errors are all output above), but the program needs to exit with status 1, so use `ExitCodeError`

Reply:
ok got the point, can we change `status`  by `code` ( more explicit I guess)  and  `error :nil` is useless
```suggestion
		return ExitCodeError{
		   exitCode: code,
		}            
```

Reply:
> ok got the point, can we change `status` by `code` ( more explicit I guess) and `error :nil` is useless

Done

Reply:
> `ExitCodeError` is to allow the program to exit with status code 1 without outputting an error message.
> 
> if return `errors.New(..)`, will output redundant error lines, like
> 
> ```shell
> ERRO[0000] network "deleteme" is in use by container ["182ea93ddc15b187419a05cb10a4b8d3cdace1fc2b8592a10b345396ce91edc7"] 
> FATA[0000] 
> ```

Could you add that in the code comment?
Or just print warnings for errs[0]...errs[n-2] and return errs[n-1]?

Reply:
> `ExitCodeError` is to allow the program to exit with status code 1 without outputting an error message.

Add that in the code comment, like this?
```go
// compatible with docker
// ExitCodeError is to allow the program to exit with status code 1 without outputting an error message.
if code != 0 {
	return ExitCodeError{
		exitCode: code,
	}
}
```

Reply:
> > `ExitCodeError` is to allow the program to exit with status code 1 without outputting an error message.
> > if return `errors.New(..)`, will output redundant error lines, like
> > ```shell
> > ERRO[0000] network "deleteme" is in use by container ["182ea93ddc15b187419a05cb10a4b8d3cdace1fc2b8592a10b345396ce91edc7"] 
> > FATA[0000] 
> > ```
> 
> Could you add that in the code comment? Or just print warnings for errs[0]...errs[n-2] and return errs[n-1]?


`return err` will result in two different outputs, like `ERRO[0000]...` and `FATA[0000]...`

I think using `ExitCodeError` here is a better choice

Reply:
> Add that in the code comment, like this?

Yes

Reply:
> > Add that in the code comment, like this?
> 
> Yes

Done